planty/posts/models.py

```diff
@@ -0,0 +1,62 @@
+from django.db import models
+from django.contrib.auth.models import User
+from datetime import datetime
+from django.core.validators import MinValueValidator
+from django.urls import reverse
+from planty.utils import crop_max_square
+from PIL import Image
+
+
+class Post(models.Model):
+
+    OUTDOOR = 'O'
+    INDOOR = 'I'
+    FLOWERS = 'F'
+    NO_CATEGORY = 'ALL'
+
+    PLANT_CHOICES = [
+        (OUTDOOR, 'Outdoor'),
+        (INDOOR, 'Indoor'),
+        (FLOWERS, 'Flower'),
+        (NO_CATEGORY, 'No category')
+    ]
+
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+    plant_category = models.CharField(
+        max_length=3,
+        choices=PLANT_CHOICES,
+        default=NO_CATEGORY,
+    )
+    title = models.CharField(max_length=30)
+    date_posted = models.DateTimeField(
+        default=datetime.now)
+    description = models.CharField(max_length=100)
+    plant_image = models.ImageField(
+        default='default_plant_image.jpg', upload_to='plant_pics')
+    price = models.FloatField(
+        default='0.00', verbose_name='Price in €',
+        validators=[MinValueValidator(0.0)])
+
+    def __str__(self):
+        return self.title
+
+    # Don’t need to provide a success_url for CreateView or UpdateView
+    # they will use get_absolute_url() on the model
+    def get_absolute_url(self):
+        return reverse('post-detail', kwargs={'pk': self.pk})
+
+    def shorten_title(self):
```

Comment:
I realize this method will be used later in some views. I would still suggest that you call this method every time a new instance is created so that you don't have to call it manually in all the views that require short title attribute on post instances. If you don't know where to call it, remember the `naloge razredi` exercises that we did yesterday :)